File: go/queue/benchmark_test.go
package queue

import (
	"testing"
)

func BenchmarkArrayQueue(b *testing.B) {
	for i := 0; i < b.N; i++ {
		arrQueue, _ := NewArrayQueue()
		for i := 0; i < 1000; i++ {
			arrQueue.Enqueue(i)
		}
		for i := 0; i < 1000; i++ {
			arrQueue.Dequeue()
		}
	}
}


func BenchmarkLoopQueue(b *testing.B) {
	for i := 0; i < b.N; i++ {
		loopQueue, _ := NewLoopQueue()
		for i := 0; i < 1000; i++ {
			loopQueue.Enqueue(i)
		}
		for i := 0; i < 1000; i++ {
			loopQueue.Dequeue()
		}
	}
}

func BenchmarkLinkedListQueue(b *testing.B) {
	for i := 0; i < b.N; i++ {
		linkedListQueue := New()
		for i := 0; i < 1000; i++ {
			linkedListQueue.Enqueue(i)
		}
		for i := 0; i < 1000; i++ {
			linkedListQueue.Dequeue()
		}
	}
}

File: js/queue/benchmarkTest.js
const ArrayQueue = require("./arrayQueue");
const LinkedListQueue = require("./linkedListQueue");
const LoopQueue = require("./loopQueue");

const arrQueue = new ArrayQueue(100000);
const llQueue = new LinkedListQueue();
const lpQueue = new LoopQueue(100000);

// 从测试结果来看，链表队列性能还是最好的，循环队列稍差，数组队列则差的无法容忍

console.time("100,000 times enqueue and dequeue ops use ArrayQueue");
for (let i = 0; i < 100000; i++) {
  arrQueue.enqueue(i);
}

for (let i = 0; i < 100000; i++) {
  arrQueue.dequeue();
}

console.timeEnd("100,000 times enqueue and dequeue ops use ArrayQueue");

console.time("100,000 times enqueue and dequeue ops use LinkedListQueue");
for (let i = 0; i < 100000; i++) {
  llQueue.enqueue(i);
}

for (let i = 0; i < 100000; i++) {
  llQueue.dequeue();
}

console.timeEnd("100,000 times enqueue and dequeue ops use LinkedListQueue");

console.time("100,000 times enqueue and dequeue ops use LoopQueue");
for (let i = 0; i < 100000; i++) {
  lpQueue.enqueue(i);
}

for (let i = 0; i < 100000; i++) {
  lpQueue.dequeue();
}

console.timeEnd("100,000 times enqueue and dequeue ops use LoopQueue");